codec-dns/src/main/java/io/netty/handler/codec/dns/DatagramDnsQueryDecoder.java

```diff
@@ -51,9 +51,16 @@ public DatagramDnsQueryDecoder(DnsRecordDecoder recordDecoder) {
     }
 
     @Override
-    protected void decode(ChannelHandlerContext ctx, DatagramPacket packet, List<Object> out) throws Exception {
+    protected void decode(ChannelHandlerContext ctx, final DatagramPacket packet, List<Object> out) throws Exception {
```

Comment:
No need to explicitly mark `DatagramPacket` as `final`.

Reply:
The parent `pom.xml` defines
```
<maven.compiler.source>1.6</maven.compiler.source>
<maven.compiler.target>1.6</maven.compiler.target>
```

And use `JDK1.6` to compile 
```
    protected void decode(ChannelHandlerContext ctx, final DatagramPacket packet, List<Object> out) throws Exception {
        final ByteBuf buf = packet.content();

        DnsMessageUtil.decodeDnsQuery(recordDecoder, buf, new DnsMessageUtil.DnsQueryFactory() {
            @Override
            public DnsQuery newQuery(int id, DnsOpCode dnsOpCode) {
                return new DatagramDnsQuery(packet.sender(), packet.recipient(), id, dnsOpCode);
            }
        });
```
need to mark `DatagramPacket` as `final`.